### PR TITLE
chore(deps): Removes mkdirp dep in favor of fs impl

### PR DESCRIPTION
**Pull request by vbudhram**
Fixes https://github.com/mozilla/fxa/issues/13406

### CODE REVIEW

File: packages/fxa-geodb/test/maxmind-db-downloader.js [hunk 1]
```diff
@@ -47,20 +47,6 @@ describe('maxmind-db-downloader', function () {
         'Directory was created'
       );
     });
-
-    it('does nothing when directory already exists', function () {
```
Comment (vbudhram) [inline packages/fxa-geodb/test/maxmind-db-downloader.js:51]: The test above does the same thing pretty much.

**Comment by clouserw** [comment]
Yes!  This is a perfect fit for this sprint.  A short patch that lets us stop using a dependency. Thanks @vbudhram

**Review by pdehaan — APPROVED** [review]

File: packages/fxa-geodb/test/maxmind-db-downloader.js (hunk 1 above)
Comment (dschom) [reply packages/fxa-geodb/test/maxmind-db-downloader.js:51]: I noticed this test breaks now. I wonder if the test should just be updated instead of removed since it does seem like an valid edge case to me. That being said, given the example usage in the readme, it doesn't really seem to matter.

**Review by dschom — APPROVED** [review]
LGTM!

File: packages/fxa-geodb/lib/maxmind-db-downloader.js [hunk 2]
```diff
@@ -38,7 +37,7 @@ var MaxmindDbDownloader = function () {
     targetDirName = targetDirName || DEFAULTS.TARGET_DIR_NAME;
     var targetDirPath = path.join(__dirname, '..', targetDirName);
     // create db directory
-    var createdTargetDirPath = mkdirp.sync(targetDirPath);
+    var createdTargetDirPath = fs.mkdirSync(targetDirPath, { recursive: true });
```
[the remark this answers is not in the document]
Comment (pdehaan) [reply packages/fxa-geodb/lib/maxmind-db-downloader.js:40]: Here was my quick test of `mkdirp` vs `fs.mkdirSync()`:

```js
const mkdirp = require("mkdirp");

const res1 = mkdirp.sync(".temp1");
console.log({res1});

const res2 = mkdirp.sync(".temp1");
console.log({res2});

// { res1: '/Volumes/Dev/github/mozilla/fxa/.temp1' }
// { res2: null }
```

```js
const fs = require("node:fs");

const res1 = fs.mkdirSync(".temp2", {recursive: true});
console.log({res1});

const res2 = fs.mkdirSync(".temp2", {recursive: true});
console.log({res2});

// { res1: '.temp2' }
// { res2: undefined }
```
